### PR TITLE
Adds event listeners for seeking

### PR DESCRIPTION
**Pull request by jaysaleh**
This PR adds event listeners for seeking video and transcript area.

### CODE REVIEW

File: src/main/webapp/view/discussion/discussion-area.js [hunk 1]
```diff
@@ -15,11 +15,12 @@
 import {timestampToString} from '../../timestamps.js';
 import DiscussionComment from './discussion-comment.js';
 import DiscussionManager from './discussion-manager.js';
-import {COMMENT_TYPE_QUESTION} from './discussion.js';
 
 export const ELEMENT_DISCUSSION =
     document.querySelector('#discussion-comments');
 
+const COMMENT_TYPE_QUESTION = 'QUESTION';
```
Comment (n-wach) [inline src/main/webapp/view/discussion/discussion-area.js:22]: Why was this moved?

File: src/main/webapp/view/discussion/discussion-area.js [hunk 2]
```diff
@@ -36,14 +38,25 @@ export default class DiscussionArea {
...
     this.#currentRootCommentElements = [];
     this.#nearestComments = [];
   }
 
+  /**
+   * Adds event listener to `eventController` allowing seeking discussion area
+   * on event broadcast.
+   */
+  addSeekingListener() {
+    this.#eventController.addEventListener((timeStamp) => {
```
Comment (n-wach) [inline src/main/webapp/view/discussion/discussion-area.js:55]: Can you rename to `timestampMs`?

File: src/main/webapp/view/lecture-view.js [hunk 3]
```diff
@@ -40,24 +44,23 @@ export default class LectureView {
...
 
-    await LectureView.video.loadVideoApi();
-    await LectureView.transcript.loadTranscript();
-    await LectureView.discussion.initialize();
+    await this.#video.loadVideoApi();
+    await this.#transcript.loadTranscript();
+    await this.#discussion.initialize();
+
+    this.#transcript.transcriptSeeker().addSeekingListener();
+    this.#discussion.addSeekingListener();
```
Comment (n-wach) [inline src/main/webapp/view/lecture-view.js:58]: Do we want to call these in their respective `.initialize` functions?

File: src/main/webapp/view/transcript/transcript-seeker.js [hunk 4]
```diff
@@ -28,7 +28,16 @@ export default class TranscriptSeeker {
...
     this.#eventController = eventController;
-    // TODO: Add the event listeners.
+  }
+
+  /**
+   * Adds event listener to `eventController` allowing seeking transcript area
+   * on event broadcast.
+   */
+  addSeekingListener() {
+    this.#eventController.addEventListener((timeStamp) => {
```
Comment (n-wach) [inline src/main/webapp/view/transcript/transcript-seeker.js:38]: Change to `timestampMs`?

File: src/main/webapp/view/discussion/discussion.js [hunk 5]
```diff
@@ -1,24 +0,0 @@
...
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-import LectureView from '../lecture-view.js';
-
-export const COMMENT_TYPE_REPLY = 'REPLY';
-export const COMMENT_TYPE_QUESTION = 'QUESTION';
-export const COMMENT_TYPE_NOTE = 'NOTE';
```
Comment (n-wach) [inline src/main/webapp/view/discussion/discussion.js:19]: Can we keep this file?  It will contain info on comment types, which is used by all other discussion javascript files.

We could also move them all to discussion-comment.js

Comment (jaysaleh) [reply src/main/webapp/view/discussion/discussion.js:19]: Sorry I think I misunderstood what you said last week, thought we were removing this file.

Added back.

File: src/main/webapp/view/discussion/discussion-area.js (hunk 1 above)
Comment (jaysaleh) [reply src/main/webapp/view/discussion/discussion-area.js:22]: Two of them aren't being used, figured it would make more sense to just have one of them here.

File: src/main/webapp/view/lecture-view.js [hunk 6]
```diff
@@ -40,24 +44,20 @@ export default class LectureView {
...
-    LectureView.video = new Video(this.#lecture, LectureView.eventController);
-    // TODO: Move TranscriptArea initialization outside of initialize()
-    // and replace string parameter with a controller object.
-    LectureView.transcript =
-        new TranscriptArea(this.#lecture, LectureView.eventController);
-    LectureView.discussion =
-        new DiscussionArea(this.#lecture, LectureView.eventController);
+    this.#eventController = new EventController();
+    this.#video = new Video(this.#lecture, this.#eventController);
+    // TODO: Move TranscriptArea initialization outside of initialize().
```
Comment (amuamushu) [inline src/main/webapp/view/lecture-view.js:49]: Should `TranscriptArea` still be moved?

Comment (jaysaleh) [reply src/main/webapp/view/lecture-view.js:49]: I'm unsure what that TODO means, I think it was added from a different PR. Do you still think we need this?

File: src/main/webapp/view/discussion/discussion-area.js (hunk 1 above)
Comment (jaysaleh) [reply src/main/webapp/view/discussion/discussion-area.js]: Moved them back to discusssion.js

Comment (n-wach) [reply src/main/webapp/view/discussion/discussion-area.js]: ah. yeah they're being used in master now

File: src/main/webapp/synchronizer.js [hunk 7]
```diff
@@ -12,27 +12,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {seekDiscussion} from './view/discussion/discussion.js';
-import TranscriptSeeker from './view/transcript/transcript-seeker.js';
-
 let lastSyncedTimeMs;
```
[the remark this answers is not in the document]
Comment (jaysaleh) [reply src/main/webapp/synchronizer.js:15]: Moved

File: src/main/webapp/synchronizer.js [hunk 8]
```diff
@@ -12,27 +12,21 @@
...
 
 const TIME_INTERVAL_MS = 100;
```
[the remark this answers is not in the document]
Comment (jaysaleh) [reply src/main/webapp/synchronizer.js:17]: Fixed

File: src/main/webapp/view/lecture-view.js [hunk 9]
```diff
@@ -40,24 +44,20 @@ export default class LectureView {
...
-    await LectureView.discussion.initialize();
+    await this.#video.loadVideoApi();
+    await this.#transcript.initialize();
+    await this.#discussion.initialize();
 
     // This is used as the `onclick` handler of the new comment area submit
     // button. It must be set after discussion is initialized.
     window.postNewComment =
-        LectureView.discussion.postNewComment.bind(LectureView.discussion);
+        this.#discussion.postNewComment.bind(this.#discussion);
```
[the remark this answers is not in the document]
Comment (jaysaleh) [reply src/main/webapp/view/lecture-view.js:60]: That makes more sense

File: src/main/webapp/view/lecture-view.js (hunk 6 above)
Comment (amuamushu) [reply src/main/webapp/view/lecture-view.js:49]: I had that there because I didn't know if `TranscriptArea` should be created here at the time.

Comment (jaysaleh) [reply src/main/webapp/view/lecture-view.js:49]: Removed

File: src/main/webapp/synchronizer.js [hunk 10]
```diff
@@ -42,20 +35,19 @@ export default class Synchronizer {
   startVideoSyncTimer() {
     setInterval(() => {
       this.sync(this.#video.getCurrentVideoTimeMs());
-    }, /* ms= */ TIME_INTERVAL_MS);
+    }, /* ms= */ Synchronizer.#TIME_INTERVAL_MS);
   }
 
   /**
-   * Calls functions that seek transcript, and discussion to
+   * Broadcasts event that seeks transcript, and discussion to
```
Comment (billie-cat) [inline src/main/webapp/synchronizer.js:42]:
```suggestion
   * Broadcasts event that seeks transcript and discussion to
```

File: src/main/webapp/synchronizer.js [hunk 11]
```diff
@@ -42,20 +35,19 @@ export default class Synchronizer {
...
    * `currentVideoTimeMs` if the `currentVideoTimeMs` changed from the last time
```
Comment (billie-cat) [inline src/main/webapp/synchronizer.js:43]:
```suggestion
   * `currentVideoTimeMs`, if the `currentVideoTimeMs` changed from the last time
```

File: src/main/webapp/view/discussion/discussion-area.js [hunk 12]
```diff
@@ -36,18 +37,35 @@ export default class DiscussionArea {
...
+    this.#eventController.addEventListener((timestampMs) => {
+      this.seek(timestampMs);
+    }, 'seek');
+  }
+
+  /**
+   * Adds event listener for seeking and initializes the discussion area by
+   * loading the current comments.
    */
   async initialize() {
```
Comment (billie-cat) [inline src/main/webapp/view/discussion/discussion-area.js:63]: I would make initialize the first method after constructor as this usually gets called after the constructor.

File: src/main/webapp/view/discussion/discussion-area.js [hunk 13]
```diff
@@ -36,18 +37,35 @@ export default class DiscussionArea {
...
+    this.addSeekingListener();
+    // This is used as the `onclick` handler of the new comment area submit
+    // button. It must be set after discussion is initialized.
+    window.postNewComment = this.postNewComment.bind(this);
```
Comment (billie-cat) [inline src/main/webapp/view/discussion/discussion-area.js:67]: Do you need to sync this PR, this code looks unrelated to this change.

File: src/main/webapp/view/discussion/discussion-area.js (hunk 12 above)
Comment (jaysaleh) [reply src/main/webapp/view/discussion/discussion-area.js:53]: fixed

File: src/main/webapp/view/discussion/discussion-area.js (hunk 13 above)
Comment (jaysaleh) [reply src/main/webapp/view/discussion/discussion-area.js:57]: Synced with base branch
